catalog/spatial/csw/spatial-csw-transformer/src/main/java/org/codice/ddf/spatial/ogc/csw/catalog/transformer/GmdTransformer.java

```diff
@@ -146,28 +223,49 @@ public Metacard transform(InputStream inputStream)
     @Override
     public Metacard transform(InputStream inputStream, String id)
             throws IOException, CatalogTransformerException {
+        return handleTransform(inputStream, id);
+    }
 
-        String xml = null;
-        XstreamPathValueTracker pathValueTracker = null;
-        xml = IOUtils.toString(inputStream);
+    private Metacard handleTransform(InputStream inputStream, String id)
+            throws CatalogTransformerException {
+        String xml;
+        XstreamPathValueTracker pathValueTracker;
+        try (ByteArrayOutputStream baos = new ByteArrayOutputStream()) {
+            IOUtils.copy(inputStream, baos);
+            byteArray = baos.toByteArray();
+
+            try (InputStream xmlSourceInputStream = getSourceInputStream()) {
+                xml = IOUtils.toString(xmlSourceInputStream);
+            }
 
-        try {
+            argumentHolder.put(XstreamPathConverter.PATH_KEY, buildPaths());
             XMLStreamReader streamReader = xmlFactory.createXMLStreamReader(new StringReader(xml));
             HierarchicalStreamReader reader = new StaxReader(new QNameMap(), streamReader);
             pathValueTracker = (XstreamPathValueTracker) xstream.unmarshal(reader,
                     null,
                     argumentHolder);
-        } catch (XStreamException | XMLStreamException e) {
+            Metacard metacard = toMetacard(pathValueTracker, id);
+            metacard.setAttribute(new AttributeImpl(Core.METADATA, xml));
+            return metacard;
+        } catch (XMLStreamException | IOException e) {
             throw new CatalogTransformerException(TRANSFORM_EXCEPTION_MSG, e);
         } finally {
             IOUtils.closeQuietly(inputStream);
         }
+    }
 
-        MetacardImpl metacard = toMetacard(pathValueTracker, id);
-
-        metacard.setAttribute(Metacard.METADATA, xml);
+/*    private void readInputStreamIntoByteArrayOutputStream(InputStream inputStream,
+            ByteArrayOutputStream baos) throws IOException {
+        byte[] buffer = new byte[BUFFER_SIZE];
+        int len;
+        while ((len = inputStream.read(buffer)) > -1) {
+            baos.write(buffer, 0, len);
+        }
+        baos.flush();
+    }*/
 
-        return metacard;
+    public InputStream getSourceInputStream() {
+        return new ByteArrayInputStream(byteArray);
     }
 
     @Override
```

Comment:
I realize this wasn't modified here but these changes made me think of this - The metacard transform should check if the metadata is already GMD XML and if so simply return that instead of serializing from the attributes.  That will maintain full fidelity of the GMD XML through the system and should be a faster transformation (since there is nothing to transform).

Here is a reference of how this is done elsewhere: https://github.com/codice/ddf/blob/master/catalog/spatial/csw/spatial-csw-transformer/src/main/java/org/codice/ddf/spatial/ogc/csw/catalog/converter/CswRecordConverter.java#L232